DataOnion/Auth/IAuthServiceStrategy.cs

```diff
@@ -0,0 +1,10 @@
+namespace DataOnion.Auth
+{
+    internal interface IAuthServiceStrategy<T>
+    {
+        Task<bool> LoginAsync(T details);
+        Task<T?> CheckSessionAsync(string id);
+        Task LogoutAsync(string id);
+        Task<bool> SetAbsoluteExpirationAsync(string id, TimeSpan expiration);
```

Comment:
i think we should just make absolute expiry part of the configuration step and not an option in this interface

Reply:
I will do this once I have clarification on the constructor.

Reply:
Disregard my constructor comment i was misunderstanding how the config step is built

Reply:
because for sessions absolute expiry is a global value in all of our use cases and it's unlikely that we will ever set 2 sessions that require different expirations

if this interface were a general redis usage layer i would be more ok with it but its specifically for sessions